addons/mail/static/src/new/core/messaging.js

```diff
@@ -1027,11 +1027,15 @@ export class Messaging {
         }
     }
 
-    openDiscussion(threadLocalId) {
+    openDiscussion(threadLocalId, replaceNewMessageChatWindow) {
         if (this.state.discuss.isActive) {
             this.setDiscussThread(threadLocalId);
         } else {
-            const chatWindow = ChatWindow.insert(this.state, { folded: false, threadLocalId });
+            const chatWindow = ChatWindow.insert(this.state, {
+                folded: false,
+                threadLocalId,
+                replaceNewMessageChatWindow,
```

Comment:
I feel like `replaceNewMessageChatWindow` should not be the concern of `insert`.
It's specific to user interaction.